### PR TITLE
Fifth draft for extended arXiv document metadata 

### PR DESCRIPTION
**Pull request by mhl10**
This fifth draft incorporates feedback from recent staff meetings and subsequent feedback from this PR.

Summary of changes:
- authors:
  - add `contributions` property. Contributions are defined by a `role` based on the same roles from the CRediT taxonomy
  - add `native_names` and `name_variants`; moved name components into parent `name` property
  - renamed Person.json to Author.json
- consolidate `submitter_dois` and `publisher_dois` under `dois` array; add `provider` and `last_updated` properties
- add `doi` property to `related_resources` to support ERC mandate; rename RelatedLink.json to RelatedResource.json
- add `retracted` to the `external_publication_status` enumeration
- add `withdrawn_by_author`, `removed`, `private` to `arxiv_status` enumeration
- relax rule for collaboration names
- remove `references` from Versions.json

Edit 1: I initially just added arXiv staff to this PR since it originated from internal discussions.
Edit 2: added members of metadata working group to PR

### CODE REVIEW

**Comment by SBBCornell** [comment]
I wanted to double check that the schema for authors/submitters will address some of the feedback we've received around internationalization, especially around multi-part first or last names. Would it account for the following situations?

"I would like to force my family name to be written 'de Roubin' but the website forces the first letter 'd' as an uppercase letter making my family name to be written 'De Roubin'."

"I am Manuel (first name) Arca Sedda (surname), but the indexing automatic method implemented in ArXiv seem to automatically assign Arca to my first name, which is not correct."

"my name is Borja Rodríguez Gálvez, with "Rodríguez Gálvez" being my surname and "Borja" my first name. The people reference to the articles I write as "B Rodríguez Gálvez ..." instead of "BR Galvez" as it is done in the arXiv database."

"my name is erreneous listed on arXiv as Xuanrui (Ray)Q. However, my first name is "Xuanrui (Ray)" and my listed last
name is "Qi", so the correct rendering of my name should be "Xuanrui (Ray) Qi".

"firstname: Ludovic, last name: Van Waerbeke. My last name is NOT just "Waerbeke" and my first name is NOT "Ludovic Van" as it seems to have recorded it."

More representative feedback is collected here: https://arxiv-org.atlassian.net/browse/AUXDH-1014.

**Comment by mhl10** [comment]
> I wanted to double check that the schema for authors/submitters will address some of the feedback we've received around internationalization, especially around multi-part first or last names. Would it account for the following situations?
> ... 
> More representative feedback is collected here: https://arxiv-org.atlassian.net/browse/AUXDH-1014.

I think most but not all of these cases are covered by the new schema. This PR adds `native_names` which handles names in non-Latin character sets. I considered also adding `preferred_name`, but I wondered if that was too semantically similar to `full_name`?

**Comment by SBBCornell** [comment]
> I considered also adding `preferred_name`, but I wondered if that was too semantically similar to `full_name`?

I could see 'preffered_name' being used by Xuanrui (Ray) Qi, where perhaps he would put Xunrui Qi under 'full_name' and Ray Qi under 'preferred name'. Another example that comes to mind is Peter K. G. Williams. His 'full_name' might spell out the K and the G, but he uses his initials to differentiate himself from other researchers named Peter Williams. I could do some research with international users and others.

Or maybe the field should be combined to 'preferred_full_name'

**Comment by jweiskoff** [comment]
Do we see `prefered_name` as being overloadable (i.e. more than one preferred name)? E.g. if someone has changed their name, but they still want it referenced (a use case could be someone who's been married multiple times and has changed their name multiple times through marriage and their publication record), might we need to consider an overloaded "AKA" to facilitate such a change?

**Comment by mhl10** [comment]
> Do we see `prefered_name` as being overloadable (i.e. more than one preferred name)? E.g. if someone has changed their name, but they still want it referenced (a use case could be someone who's been married multiple times and has changed their name multiple times through marriage and their publication record), might we need to consider an overloaded "AKA" to facilitate such a change?

I don't think we would want to allow more than one `preferred_name`, but we could support what you're describing with `previous_names`. INSPIRE is supporting this in their schema for authors:

https://github.com/inspirehep/inspire-schemas/blob/master/inspire_schemas/records/authors.yml

**Comment by SBBCornell** [comment]
> INSPIRE is supporting this in their schema for authors

I see they are also supporting "name_variants". That sounds like it might address some of our user complaints too. For example _"At the moment I have 12 articles on arXiv, but the system considers me to be two different authors: 1912.00056 and 1908.11181 are listed as authored by "Price, A E" whereas 1707.09120 and 1612.03293 are listed as written by "Elvey-Price, A". In every case my name should be listed as "Elvey Price, A". I hope that this can be fixed."_

And _"I tried to ask the co-authors to insert me in only one way (just using Dalla Bontà or Dalla Bonta`), but my collaborators, coming from different countries, were not able to follow my suggestion, so I assume that I will keep on appearing on the arxiv in different ways."_

**Comment by jweiskoff** [comment]
The other variants cold also be extended into (un)hyphenated variants for multi-word names as well.

**Comment by jimentwood** [comment]
I wonder about relying on orcid for variations and having a simple schema for arXiv...
________________________________

File: schema-extended/validate.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "Funding.json",
+    "Keyword.json",
+    "LanguageCode.json",
+    "License.json",
+    "Author.json",
+    "References.json",
+    "RelatedLink.json",
+    "Role.json",
+    "Taxonomy.json",
+    "Version.json"
```
Comment (michamos) [inline schema-extended/validate.py]: @mhl10 did you forget to commit some of these? or are they still WIP?

Comment (mhl10) [reply schema-extended/validate.py]: @michamos they should all be there, in the same directory as the script.

Comment (michamos) [reply schema-extended/validate.py]: I can't find `Funding` for example (using the `jump to` menu at the top)

Comment (mhl10) [reply schema-extended/validate.py]: Strange, I'm not sure why that isn't working. Here is `Funding.json` in any case:
https://github.com/arXiv/arxiv-canonical/blob/arxiv-extended-metadata/schema-extended/Funding.json

Comment (michamos) [reply schema-extended/validate.py]: my bad, this is only the latest unmerged diff, unchanged files don't appaear here. Thanks!

File: schema-extended/RelatedResource.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  "properties": {
+    "identifier": {
+      "oneOf": [{
+          "type": "string",
+          "format": "uri"
+        },
+        {
+          "$ref": "Common.json#/definitions/doi"
+        }
+      ]
```
Comment (michamos) [inline schema-extended/RelatedResource.json]: I'd recommend against simply overloading the `identifier` to mean `url or doi`, as it's often very useful to know what kind of identifier you're dealing with (for example, if you display this somewhere you'll need to turn the DOI into a URL).

Comment (mhl10) [reply schema-extended/RelatedResource.json]: That's a good point. I'll revise--thanks!

**Review by oxmynx — COMMENTED** [review]
No comments on these changes but one additional requirement that I was made aware of during the creation of the ERC proposal: They are going to introduce Grant DOIs. We need to allow both the grant number (original internal ID) as well as Grant DOIs to be added. Both can exist in parallel and are obviously not mandatory.

**Pull request merged**